AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveQueue.swift

```diff
@@ -0,0 +1,116 @@
+    }
+    required init(_ modelSchemas: [ModelSchema]) {
+        self.reconcileAndSaveQueueSubject = PassthroughSubject<ReoncileAndSaveQueueEvent, DataStoreError>()
+        self.reconcileAndSaveQueue = OperationQueue()
+        reconcileAndSaveQueue.name = "com.amazonaws.DataStore.reconcile"
+        reconcileAndSaveQueue.maxConcurrentOperationCount = 1
+        reconcileAndSaveQueue.underlyingQueue = DispatchQueue.global()
+        reconcileAndSaveQueue.isSuspended = false
+
+        for model in modelSchemas {
+            modelReconcileAndSaveOperations[model.name] = [UUID: ReconcileAndLocalSaveOperation]()
+        }
+        self.reconcileAndLocalSaveOperationSinks = AtomicValue(initialValue: Set<AnyCancellable?>())
+    }
+
+    func addOperation(_ operation: ReconcileAndLocalSaveOperation, modelName: String) {
+        serialQueue.async {
+            var reconcileAndLocalSaveOperationSink: AnyCancellable?
+
+            reconcileAndLocalSaveOperationSink = operation.publisher.sink { [weak self] _ in
+                self?.reconcileAndLocalSaveOperationSinks.with { $0.remove(reconcileAndLocalSaveOperationSink) }
+                self?.modelReconcileAndSaveOperations[modelName]?[operation.id] = nil
+                self?.reconcileAndSaveQueueSubject.send(.operationRemoved(id: operation.id))
+            } receiveValue: { _ in }
+
+            self.reconcileAndLocalSaveOperationSinks.with { $0.insert(reconcileAndLocalSaveOperationSink) }
+            self.modelReconcileAndSaveOperations[modelName]?[operation.id] = operation
+            self.reconcileAndSaveQueue.addOperation(operation)
+            self.reconcileAndSaveQueueSubject.send(.operationAdded(id: operation.id))
+        }
+    }
+
+    func cancelOperations(modelName: String) {
+        serialQueue.async {
+            if let operations = self.modelReconcileAndSaveOperations[modelName] {
+                operations.values.forEach { operation in
+                    operation.cancel()
+                }
+            }
+            self.modelReconcileAndSaveOperations[modelName]?.removeAll()
+            self.reconcileAndSaveQueueSubject.send(.cancelledOperations(modelName: modelName))
+        }
+    }
+
+    func cancelAllOperations() {
+        serialQueue.async {
+            self.reconcileAndSaveQueue.cancelAllOperations()
+            for (modelName, _) in self.modelReconcileAndSaveOperations {
+                self.modelReconcileAndSaveOperations[modelName]?.removeAll()
+                self.reconcileAndSaveQueueSubject.send(.cancelledOperations(modelName: modelName))
+            }
+        }
+    }
+
+    // This method blocks the current thread by not executing the work on the serial queue since underlying operation
+    // queue's `waitUntilAllOperationsAreFinished()` behaves the same way. See the following link for more details:
+    // https://developer.apple.com/documentation/foundation/operationqueue/1407971-waituntilalloperationsarefinishe
```

Comment:
Add an additional note that this is only used for `reset`; we wouldn't really want to do this in a normal flow I think?